src/index.ts

```diff
@@ -221,8 +221,8 @@ export class Paginator {
         delete query.maxApiCalls;
       }
 
-      // maxResults is the user specified a limit.
-      if (callback && (maxResults !== -1 || query.autoPaginate === false)) {
```

Comment:
I've been thinking about this change a bit and I have some concerns. Since we remove the callback check I think this means that this logic will now apply to any object streams. Hopefully @stephenplusplus can chime in here but I think that this means setting `maxResults` on an object stream method will cause it to come in under the desired amount since now we'll be flagging `autoPaginate` to false.

Reply:
For a follow up thought, since callback was previously used to determine if we wanted a stream or not, maybe we should create an internal `streaming` flag that we check in place of this. 

```typescript
parseArguments: (_args: any[], streaming: boolean);
```

Reply:
`autoPaginate` is only used by `paginator.extend` to determine whether we wanted a stream or not. 
`autoPaginate` could get flagged to `false` only by user setting `maxResults`/`pageSize` to a value or setting `autoPaginate` to `false`. Previously all methods came with `callback`s (even if user did not provide one). Now since we are introducing promise based methods, `callback` becomes optional, and we might run in an issue of not flagging `autoPaginate` to false when it is needed.

Reply:
> autoPaginate is only used by paginator.extend to determine whether we wanted a stream or not.

This isn't quite accurate,`autoPaginate` is used to determine if we should make multiple requests. A separate method (`Paginator.streamify()`) is used for generating streams, which also calls `parseArguments_`.

> Previously all methods came with callbacks (even if user did not provide one). 

This is true for callback/promise methods, but `callback` has always been undefined when called from `Paginator.streamify()`.

> Now since we are introducing promise based methods, callback becomes optional, and we might run in an issue of not flagging autoPaginate to false when it is needed.

That's why I suggested adding a second boolean parameter to determine if the user wants a stream or not. Setting said parameter to `false` inside of `Paginator.extend()` and `true` inside of `Paginator.streamify()`.

Reply:
Actually, I just realized `runAsStream_` ignores `autoPaginate` so I guess it doesn't matter. Feel free to ignore this and sorry about the confusion!